### PR TITLE
[INFRA-744] Update Ptolemy

### PR DESCRIPTION
**Pull request by xasos**
- Update to ES6
- New name field for EPSG codes
- Add new tests + error fields
- Add new supported formats

### CODE REVIEW

File: error.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+class InvalidFormatError extends ExtendableError {
+  constructor(m) {   
+    super(m);
+  }
+}
+
+module.exports = {
+	InvalidSRIDError: InvalidSRIDError,
```
Comment (dremonkey) [inline error.js]: spacing

File: index.js [hunk 2]
```diff
@@ -37,26 +56,44 @@ function validateEPSG (srid) {
...
-};
+	
+	var requestURL = BASE_URL + epsg + '.' + format;
+  // Note: We parse the SRID's XML to get the Site Name (avoids scraping)
+	var nameURL = BASE_URL + epsg + '.xml';
+  
+	return promisfiedGet(requestURL, {timeout: 4000})
+	.then((res) => {
+    if (res.statusCode != 200) {
+      return Bluebird.reject(new Error(res.statusCode));
```
Comment (dremonkey) [inline index.js]: Create a new error and add the response body. Status code alone is pretty useless.

File: index.js [hunk 3]
```diff
@@ -37,26 +56,44 @@ function validateEPSG (srid) {
...
+      try {
+        returnObj.name = res.body["gml:ProjectedCRS"]["gml:srsName"];
+      }
+      catch (e) {
+        returnObj.name = res.body["gml:GeographicCRS"]["gml:srsName"];
+      }
+
+      return returnObj;
+    });
+	}).catch((e) => {
```
Comment (dremonkey) [inline index.js]: spacing

File: index.js [hunk 4]
```diff
@@ -37,26 +56,44 @@ function validateEPSG (srid) {
...
   }
-};
+	
+	var requestURL = BASE_URL + epsg + '.' + format;
+  // Note: We parse the SRID's XML to get the Site Name (avoids scraping)
+	var nameURL = BASE_URL + epsg + '.xml';
+  
+	return promisfiedGet(requestURL, {timeout: 4000})
+	.then((res) => {
+    if (res.statusCode != 200) {
```
Comment (dremonkey) [inline index.js]: spacing

**Comment by dremonkey** [comment]
Looks pretty good. There are a number of spacing issues that should be fixed. Biggest issue is with the way non-200 responses are handled.

**Comment by xasos** [comment]
@dremonkey, updated acc. to your comments

**Pull request merged**